File: Manual/LightField/Add-in and Automation SDK/Samples/VisualBasic Add-Ins/Viewer Sample/ViewerSample.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Windows
Imports System.Windows.Controls
Imports System.Windows.Markup
Imports System.AddIn
Imports System.IO
Imports System.AddIn.Pipeline
Imports PrincetonInstruments.LightField.AddIns

'''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
'''''''''' Demonstrate opening a file through LightField                             ''''''''''
'''''''''' and the putting it into custom WPF view as well as using a LightFieldView ''''''''''
'''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
Public Class AddinViewerSample
    Inherits AddInBase

    Private control_ As ViewerSample

    Public ReadOnly Property UISupport As UISupport
        Get
            Return UISupport.ExperimentView
        End Get
    End Property

    Public Sub Activate(ByVal app As ILightFieldApplication)
        'Capture interface
        LightFieldApplication = app
        'Build your controls
        control_ = New ViewerSample(LightFieldApplication)
        ExperimentViewElement = control_
    End Sub

    Public Sub Deactivate()
    End Sub

    Public Overrides ReadOnly Property UIExperimentViewTitle As String
        Get
            Return "Viewer Sample"
        End Get
    End Property

End Class

File: Manual/LightField/Add-in and Automation SDK/Samples/CSharp Add-Ins/Viewer Sample/ViewerSample.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Windows;
using System.Windows.Controls;
using System.AddIn;
using System.AddIn.Pipeline;

using PrincetonInstruments.LightField.AddIns;

namespace LightFieldAddInSamples
{
    ///////////////////////////////////////////////////////////////////////////
    ///////////////////////////////////////////////////////////////////////////
    ///////////////////////////////////////////////////////////////////////////
    ///////////////////////////////////////////////////////////////////////////
    ///////////////////////////////////////////////////////////////////////////
    [AddIn("Viewer Sample (C#)", 
            Version = "1.0.0",
            Publisher = "Teledyne Princeton Instruments", 
            Description = "Shows how to create your own view of data, a report, or any custom view from within LightField \nThis example allows you to open a file through LightField and then converts it to a bitmap and displays it")]
    [QualificationData("IsSample", "True")]
    public class AddinViewerSample : AddInBase, ILightFieldAddIn
    {
        private ViewerSample control_;
        ///////////////////////////////////////////////////////////////////////
        public UISupport UISupport { get { return UISupport.ExperimentView; } }
        ///////////////////////////////////////////////////////////////////////
        public void Activate(ILightFieldApplication app)
        {
            // Capture Interface
            LightFieldApplication = app;

            // Build your controls
            control_ = new ViewerSample(LightFieldApplication);
            ExperimentViewElement = control_;
        }
        ///////////////////////////////////////////////////////////////////////
        public void Deactivate() { }                
        ///////////////////////////////////////////////////////////////////////
        public override string UIExperimentViewTitle { get { return "Viewer Sample"; } }
    }
}
